backend/api/firebase_auth.py

```
@@ -0,0 +1,64 @@
+    """
+    def authenticate(self, request):
+        # Get the auth header
+        auth_header = request.META.get('HTTP_AUTHORIZATION')
+        if not auth_header:
+            return None
+
+        # Extract the token
+        try:
+            id_token = auth_header.split(' ').pop()
+        except IndexError:
+            raise exceptions.AuthenticationFailed('Invalid token format')
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)